From 081f3618a795de7428a3a10e57c2952e028cfe0d Mon Sep 17 00:00:00 2001
From: Andrii Nakryiko <andrii.nakryiko@gmail.com>
Date: Mon, 3 Feb 2020 08:52:26 +0100
Subject: [PATCH] dwarves: Add -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 to
 build libbpf

We need that as the fix in upstream libbpf is in the
tools/lib/bpf/Makefile, that isn't used when building libbpf as part of
pahole, see:

  "Makefile: back-port _FILE_OFFSET_BITS=64 and _LARGEFILE64_SOURCE to Makefile"
  https://github.com/libbpf/libbpf/commit/4a50ceb0436f17a658b48fc03b8d72fb97522847

That refers to this in the kernel sources:

  71dd77fd4bf7 ("libbpf: use LFS (_FILE_OFFSET_BITS) instead of direct mmap2 syscall")

Cc: Andrii Nakryiko <andrii.nakryiko@gmail.com>
Tested-by: Arnaldo Carvalho de Melo <acme@redhat.com>
Cc: Alexei Starovoitov <ast@kernel.org>
Cc: Jiri Olsa <jolsa@kernel.org>
Cc: Julia Kartseva <hex@fb.com>
Cc: Yonghong Song <yhs@fb.com>
Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4d23dbe..3e55a40 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,6 +74,8 @@ if (NOT HAVE_REALLOCARRAY_SUPPORT)
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DCOMPAT_NEED_REALLOCARRAY")
 endif()
 
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64")
+
 file(GLOB libbpf_sources "lib/bpf/src/*.c")
 add_library(bpf OBJECT ${libbpf_sources})
 set_property(TARGET bpf PROPERTY POSITION_INDEPENDENT_CODE 1)
-- 
2.25.1

